apps/meteor/ee/server/models/raw/LivechatRooms.ts

```diff
@@ -726,4 +733,33 @@ export class LivechatRoomsRawEE extends LivechatRoomsRaw implements ILivechatRoo
 			...extraQuery,
 		});
 	}
+
+	updateMergedContactIds(
+		contactIdsThatWereMerged: ILivechatContact['_id'][],
+		newContactId: ILivechatContact['_id'],
+		options?: UpdateOptions,
+	): Promise<UpdateResult | Document> {
+		return this.updateMany({ contactId: { $in: contactIdsThatWereMerged } }, { $set: { contactId: newContactId } }, options);
+	}
+
+	findClosedRoomsByContactAndSourcePaginated({
+		contactId,
+		source,
+		options = {},
+	}: {
+		contactId: string;
+		source?: string;
+		options?: FindOptions;
+	}): FindPaginated<FindCursor<IOmnichannelRoom>> {
+		return this.findPaginated<IOmnichannelRoom>(
+			{
+				contactId,
+				closedAt: { $exists: true },
+				...(source && {
+					$or: [{ 'source.type': new RegExp(escapeRegExp(source), 'i') }, { 'source.alias': new RegExp(escapeRegExp(source), 'i') }],
```

Comment:
is this required to be a `$regex`? from what I could see `source.type` is a fixed list:
https://github.com/RocketChat/Rocket.Chat/blob/7638d07bb8440a79c4fc204ba9c836c6302247b3/packages/core-typings/src/IRoom.ts#L161-L168

and according to the type definition comment, `source.alias` is just for "post analytical purposes", should we filter for it as well?

Reply:
We'll check this [here](https://rocketchat.atlassian.net/browse/SCI-176?atlOrigin=eyJpIjoiZWY5MTM0OGYzYjVmNGM1Y2FiNGU2ZmYwZDU5NzI4M2IiLCJwIjoiaiJ9)